src/india_api/internal/service/server.py

```diff
@@ -133,3 +173,185 @@ class GetHealthResponse(BaseModel):
+    resample_minutes: Optional[int] = None,
+) -> GetHistoricGenerationResponse:
+    """Function for the historic generation route."""
+    values: list[ActualPower] = []
+
+    try:
+        if source == "wind":
+            values = db.get_actual_wind_power_production_for_location(location=region)
+        elif source == "solar":
+            values = db.get_actual_solar_power_production_for_location(location=region)
+    except Exception as e:
+        raise HTTPException(
+            status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
+            detail=f"Error getting {source} power production: {e}",
+        ) from e
+
+    if resample_minutes is not None:
+        values = resample_generation(values=values, internal_minutes=resample_minutes)
+
+    return GetHistoricGenerationResponse(
+        values=[y.to_timezone(tz=local_tz) for y in values if y.Time < dt.datetime.now(tz=dt.UTC)],
+    )
+
+
+class GetForecastGenerationResponse(BaseModel):
+    """Model for the forecast generation endpoint response."""
+
+    values: list[PredictedPower]
+
+
+@server.get(
+    "/{source}/{region}/forecast",
+    tags=["Forecast Routes"],
+    status_code=status.HTTP_200_OK,
+)
+def get_forecast_timeseries_route(
+    source: ValidSourceDependency,
+    region: str,
+    db: DBClientDependency,
+    auth: dict = Depends(auth),
+    # TODO: add auth scopes
+    forecast_horizon: ForecastHorizon = ForecastHorizon.day_ahead,
+    forecast_horizon_minutes: Optional[int] = None,
+) -> GetForecastGenerationResponse:
+    """Function for the forecast generation route.
+
+    Args:
+        source: The source of the forecast, this is current wind or solar.
+        region: The region to get the forecast for.
+        forecast_horizon: The time horizon to get the data for. Can be 'latest', 'horizon' or 'day ahead'
+        forecast_horizon_minutes: The number of minutes to get the forecast for. forecast_horizon must be 'horizon'
+    """
+    values: list[PredictedPower] = []
+
+    try:
+        if source == "wind":
+            values = db.get_predicted_wind_power_production_for_location(
+                location=region,
+                forecast_horizon=forecast_horizon,
+                forecast_horizon_minutes=forecast_horizon_minutes,
+            )
+        elif source == "solar":
+            values = db.get_predicted_solar_power_production_for_location(
+                location=region,
+                forecast_horizon=forecast_horizon,
+                forecast_horizon_minutes=forecast_horizon_minutes,
+            )
+    except Exception as e:
+        raise HTTPException(
+            status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
+            detail=f"Error getting {source} power production: {e}",
+        ) from e
+
+    return GetForecastGenerationResponse(
+        values=[y.to_timezone(tz=local_tz) for y in values],
+    )
+
+
+@server.get(
+    "/{source}/{region}/forecast/da-csv", tags=["Forecast Routes"], response_class=FileResponse
```

Comment:
Wondering if the `da-csv` route segment is a bit specific, if someone wanted the full forecast, or forecast + generation, would that be extra duplicate routes? Could do just `/{source}/{region}/forecast/da-csv` with an optional param with default value like `type = "day-ahead"` or something..?

Reply:
At the moment the "tomorrow only" code is in the `format_csv` function, so that might need rejigging as well if so though

Reply:
So actually, the DA thing here is two things, the forecast horizon and the forecast values filter... just don't want to start using this endpoint and then almost immediately be adding another/more for other similar things

Reply:
yea its very specific for the use case right now. 

Perhaps I can change it to `/{source}/{region}/forecast/day-ahead-csv` and then if we want to expand it to different forecast horizons, then we can

Reply:
or we do `/{source}/{region}/forecast/csv` and then we add forecast horizon as a parameter

I think its ok if we limit it to DA, and we can expand it later on